### PR TITLE
Annotate nft_puzzles.py

### PR DESCRIPTION
**Pull request by AmineKhaldi**

### CODE REVIEW

File: chia/wallet/nft_wallet/nft_puzzles.py [hunk 1]
```diff
@@ -58,11 +58,7 @@ def create_nft_layer_puzzle_with_curry_params(
 
 def create_full_puzzle_with_nft_puzzle(singleton_id: bytes32, inner_puzzle: Program) -> Program:
     if log.isEnabledFor(logging.DEBUG):
-        log.debug(
-            "Creating full NFT puzzle with inner puzzle: \n%r\n%r",
-            singleton_id,
-            inner_puzzle.get_tree_hash(),
-        )
+        log.debug("Creating full NFT puzzle with inner puzzle: \n%r\n%r", singleton_id, inner_puzzle.get_tree_hash())
```
Comment (arvidn) [inline chia/wallet/nft_wallet/nft_puzzles.py:61]: this `get_tree_hash()` is probably pretty expensive. I believe we call this regardless of whether debug logging is enabled

**Review by arvidn — APPROVED** [review]
I would guess that using a pragma comment to suppress a type warning is better than `cast()` because once the typing is fixed for `clvm`, mypy will warn about the pointless pragma and we can remove it. Will mypy warn about a pointless cast?

**Comment by AmineKhaldi** [comment]
> I would guess that using a pragma comment to suppress a type warning is better than `cast()` because once the typing is fixed for `clvm`, mypy will warn about the pointless pragma and we can remove it. Will mypy warn about a pointless cast?

Yep, `[redundant-cast]`.

File: chia/wallet/nft_wallet/nft_puzzles.py (hunk 1 above)
Comment (AmineKhaldi) [reply chia/wallet/nft_wallet/nft_puzzles.py:61]: We have it under the condition `if log.isEnabledFor(logging.DEBUG):` so that should be fine.

**Pull request merged**